hazelcast/src/main/java/com/hazelcast/internal/server/tcp/TcpServerConnectionManagerBase.java

```diff
@@ -175,28 +255,26 @@ public void onConnectionClose(TcpServerConnection connection, Throwable cause, b
             }
 
             Address remoteAddress = connection.getRemoteAddress();
+            Plane plane = null;
             if (remoteAddress != null) {
                 int planeIndex = connection.getPlaneIndex();
                 if (planeIndex > -1) {
-                    Plane plane = planes[connection.getPlaneIndex()];
-                    plane.connectionsInProgress.remove(remoteAddress);
-                    plane.connectionMap.remove(remoteAddress);
+                    plane = planes[connection.getPlaneIndex()];
+                    plane.removeConnection(connection);
                     fireConnectionRemovedEvent(connection, remoteAddress);
                 } else {
                     // it might be the case that the connection was closed quickly enough
                     // that the planeIndex was not set. Instead look for the connection
                     // by remoteAddress and connectionId over all planes and remove it wherever found.
                     boolean removed = false;
-                    for (Plane plane : planes) {
-                        plane.connectionsInProgress.remove(remoteAddress);
+                    for (Plane p : planes) {
+                        if (p.removeConnectionInProgress(remoteAddress)) {
+                            plane = p;
+                        }
                         // not using removeIf due to https://bugs.java.com/bugdatabase/view_bug.do?bug_id=8078645
-                        Iterator<TcpServerConnection> connections = plane.connectionMap.values().iterator();
-                        while (connections.hasNext()) {
-                            TcpServerConnection cxn = connections.next();
-                            if (cxn.getConnectionId() == connection.getConnectionId()) {
-                                connections.remove();
-                                removed = true;
-                            }
+                        if (p.removeConnectionsWithId(connection.getConnectionId())) {
```

Comment:
The comment in above line should be moved inside `removeConnectionsWithId` implementation, because now it is out of context.